lib/Doctrine/ORM/Mapping/Driver/XmlDriver.php

```diff
@@ -861,6 +862,21 @@ protected function loadMappingFile($file)
         return $result;
     }
 
+    private function assertValid(string $file): void
+    {
+        $backedUpErrorSetting = libxml_use_internal_errors(true);
+        $document = new \DOMDocument();
+        $document->load($file);
+        if (!$document->schemaValidate(__DIR__ . '/../../../../../doctrine-mapping.xsd')) {
+            $errors = libxml_get_errors();
+            libxml_clear_errors();
+            libxml_use_internal_errors(false);
```

Comment:
These two directives could be put into a more DRY `finally` block, and use `$backedUpErrorSetting`.

Reply:
I lost my train of thought it seems